packages/core/src/telemetry.ts

```diff
@@ -137,16 +140,32 @@ export const telemetrySender = (
           { timeout: httpRequestTimeout },
         )
         queuedEvents = []
+        consecutiveRetryCount = 0
       } catch (e) {
         log.debug(`failed sending telemetry events: ${e}`)
+        if (e.response) {
+          const { status } = e.response
+          if (status >= 400 && status < 500) {
+            log.debug('telemetry http response status: %d, giving up on sending events', status)
+            stopped = true
+            return
+          }
+        }
+        consecutiveRetryCount += 1
+        if (consecutiveRetryCount === MAX_CONSECUTIVE_RETRIES) {
+          log.debug('reached maximum retries: %d, giving up on sending events', MAX_CONSECUTIVE_RETRIES)
+          stopped = true
+        }
       }
     }
   }
 
   const start = (): void => {
-    timer = setTimeout(() => {
-      flush()
-      start()
+    timer = setTimeout(async () => {
+      await flush()
+      if (!stopped) {
+        start()
+      }
```

Comment:
with this implementation calling `start` after `stop` will have a really weird behavior - it will stay "stopped" but it will send one "round" of events (will call flush once).

if we don't want to support re-starting after `stop`, the `!stopped` check should probably go outside
```suggestion
    if (!stopped) {
      timer = setTimeout(() => flush().then(() => start())), flushInterval)
    }
```

if we do want to support re-starting after `stop`, then the `start` function should have `stopped = false` in the beginning

Reply:
another option is to have the `if (!stopped)` part cover the call to `flush` as well